File: src/main/java/eu/mshade/mwork/binarytag/BinaryTagMarshal.java
package eu.mshade.mwork.binarytag;

public interface BinaryTagMarshal<T> {

    BinaryTag<?> serialize(BinaryTagDriver binaryTagDriver, T t) throws Exception;

    T deserialize(BinaryTagDriver binaryTagDriver, BinaryTag<?> binaryTag) throws Exception;
}

File: src/main/java/eu/mshade/mwork/binarytag/BinaryTagMarshal.kt
package eu.mshade.mwork.binarytag

interface BinaryTagMarshal<T> {

    fun serialize(binaryTagDriver: BinaryTagDriver, t: T): BinaryTag<*>

    fun deserialize(binaryTagDriver: BinaryTagDriver, binaryTag: BinaryTag<*>): T
}